Ensure all modules in the dict are instance instead of class

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72470
* __->__ #72615

1. Add unit test to ensure the key in `ALL_MODULES` is an instance
2. In `generate_models.py`, add a check to ensure the key is an instance of `torch.nn.Module`.



Differential Revision: [D34116938](https://our.internmc.facebook.com/intern/diff/D34116938/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34116938/)!

Differential Revision: [D34116938](https://our.internmc.facebook.com/intern/diff/D34116938)

Commit:
Ensure all modules in the dict are instance instead of class



Differential Revision: [D34116938](https://our.internmc.facebook.com/intern/diff/D34116938/)

[ghstack-poisoned]

Commit:
Update on "Ensure all modules in the dict are instance instead of class"


1. Add unit test to ensure the key in `ALL_MODULES` is an instance
2. In `generate_models.py`, add a check to ensure the key is an instance of `torch.nn.Module`.



Differential Revision: [D34116938](https://our.internmc.facebook.com/intern/diff/D34116938/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34116938/)!

Differential Revision: [D34116938](https://our.internmc.facebook.com/intern/diff/D34116938)

[ghstack-poisoned]

Commit:
Update on "Ensure all modules in the dict are instance instead of class"


1. Add unit test to ensure the key in `ALL_MODULES` is an instance
2. In `generate_models.py`, add a check to ensure the key is an instance of `torch.nn.Module`.



Differential Revision: [D34116938](https://our.internmc.facebook.com/intern/diff/D34116938/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34116938/)!

Differential Revision: [D34116938](https://our.internmc.facebook.com/intern/diff/D34116938)

[ghstack-poisoned]

tugsbayasgalan commented on test/jit/fixtures_srcs/test_upgrader_models_generation.py:
```diff
@@ -0,0 +1,20 @@
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+
+class TestUpgraderModelGeneration(TestCase):
```
Isn't this redundant since we are basically checking the same thing as the original file.

cccclai replied:
It's for different purposes. 

- Unit test can run per CI to protect the model generation process, so it can prevent committing changes to this process. 

- In the meanwhile, the error message in `generate_models.py` is more like, user makes changes in `generated_models.py` and the error message can provide immediate feedback about what is wrong. 

It's for addressing the feedback from @gmagogsfm in https://github.com/pytorch/pytorch/pull/72313:
> Do we have tests to verify that the key to this dictionary is a TestCase instance not a TestCase class? I think one of the common mistake is forgetting those parenthese.

tugsbayasgalan replied:
Oh got it

Commit:
Update on "Ensure all modules in the dict are instance instead of class"


1. Add unit test to ensure the key in `ALL_MODULES` is an instance
2. In `generate_models.py`, add a check to ensure the key is an instance of `torch.nn.Module`.



Differential Revision: [D34116938](https://our.internmc.facebook.com/intern/diff/D34116938/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34116938/)!

Differential Revision: [D34116938](https://our.internmc.facebook.com/intern/diff/D34116938)

[ghstack-poisoned]
